polars/polars-io/src/lib.rs

```diff
@@ -134,7 +134,14 @@ pub(crate) fn finish_reader<R: ArrowReader>(
             }
         }
     }
-    let mut df = accumulate_dataframes_vertical(parsed_dfs)?;
+    let mut df = {
+        if parsed_dfs.is_empty() {
```

Comment:
I think this branch should take the `arrow_schema :&ArrowSchema` and use that to produce the empty columns.

Something like this:

```rust
    let empty_cols = arrow_schema.fields.iter().map(|fld| {
        Series::try_from((fld.name.as_str(), Arc::from(new_empty_array(fld.data_type.clone())) as ArrayRef))
    }).collect::<Result<_>>()?;
    DataFrame::new(empty_cols)
```